src/ts/ConvivaAnalytics.ts

```diff
@@ -243,32 +250,13 @@ export class ConvivaAnalytics {
     return this.sessionKey !== Conviva.Client.NO_SESSION_KEY;
   }
 
-  private onSourceLoaded = () => {
-    if (this.isAd) {
-      // Ignore ON_SOURCE_LOADED events during ad playback, because that's just an ad being temporarily loaded
-      // instead of the actual source.
+  private onPlaybackStateChanged = (event?: any) => {
+    this.debugLog('reportplaybackstate', event);
+    if (event && event.issuer === 'IMA') {
```

Comment:
is this check really save? 

I guess this check should prevent the playback state update from Ad Events right? 

So what if the issuer is not `IMA` for example when the Ad does not get handled by the IMA SDK?

Reply:
No it isn't. It's just a naive approach to silence playback state changes during an ad. I'm not totally sure why those play events are even triggered. But at least for the case we play an IMA ad it is working.

If this check is not working it's not the end of the world because conviva is suppress all playback state change events between `adStarted` and `adEnded` by default. But anyway they should not be send.